doc/source/hclust.rst

```diff
@@ -0,0 +1,41 @@
+Hierarchical Clustering
+========================
+
+`Hierarchical clustering <https://en.wikipedia.org/wiki/Hierarchical_clustering>`_ algorithms build a dendrogram of nested clusters by repeatedly merging or splitting clusters.
+
+**Functions**
+
+.. function:: hclust(D, method)
+
+	Perform hierarchical clustering on distance matrix D with specified method.
+
+	:param D: The pairwise distance matrix. ``D[i,j]`` is the distance between points ``i`` and ``j``.
+	:param method: A Symbol specifying how distance is measured between clusters (which is used to determine which clusters to merge on each iteration). Valid methods are ``:single``, ``:average``, or ``:complete``.
+		
+	- ``:single``: cluster distance is equal to the minimum distance between any of the members
+	- ``:average``: cluster distance is equal to the mean distance between any of the cluster's members
+	- ``:complete``: cluster distance is equal to the maximum distance between any of the members. 
+
+The function returns an object of type `Hclust` with the fields
+	 - ``merge`` the clusters merged in order.  Leafs are indicated by negative numbers
+	 - ``height`` the distance at which the merges take place
+	 - ``order`` a preferred grouping for drawing a dendogram.
+	 - ``method`` the name of the clustering method. 
+
+Example:
+
+	.. code-block:: julia
+
+		D = rand(1000,1000)
+		D += D'  # symmetric distance matrix (optional)
+		result = hclust(D, :single)
+
+.. function:: cutree(result; h, k)
```

Comment:
Hmm, here `h` and `k` are swapped in comparison to the source code.
Also what is missing is the logic describing when `cutree()` uses `h` and when `k`.

cc @davidavdav